### PR TITLE
Disabled Dropdown Options fix

### PR DESCRIPTION
**Pull request by clintcs**
## 🚀 Description

> ### Patch
> Dropdown when filterable and all but one Dropdown Option is disabled no longer sets the value of its input field to "Select All" when the enabled Dropdown Option is selected.

## 📋 Checklist

- I have followed the [Contributing Guidelines](https://github.com/crowdstrike/glide-core/blob/main/CONTRIBUTING.md).
- I have added tests to cover new or updated functionality.
- I have added or updated Storybook stories.
- I have [localized](https://github.com/CrowdStrike/glide-core/blob/main/CONTRIBUTING.md#translations-and-static-strings) new strings.
- I have followed the [ARIA Authoring Practices Guide](https://www.w3.org/WAI/ARIA/apg/patterns/) or met with the Accessibility Team.
- I have included a [changeset](https://github.com/CrowdStrike/glide-core/blob/main/CONTRIBUTING.md#versioning-a-package).
- I have scheduled a design review.
- I have reviewed the Storybook and Visual Test Report links below.

## 🔬 Testing

I intentionally omitted a test. I suspect that'll be contentious, and I don't know if I can describe my intuition well. But something doesn't seem right to me about adding tests for edge-case bugs whose root cause is structural. 

Part of it, I think, has to do with the fact the space of possible tests—covering every situation—is nearly infinite. For example, say we got a bug report where Dropdown's has three options and only the last option couldn't be disabled—or only its last option couldn't be selected. 

It's not hard to imagine hundreds of situations like this. Yet we don't test for them because they (like this bug) would be indicative of a deeper problem—and testing for them would produce an almost unlimited number of tests.

Anyway, I'm happy to add a test if you disagree or think differently. Just let me know. Either way, if you want to test the fix manually:

1. Navigate to Dropdown in Storybook.
2. Make Dropdown filterable.
3. Use DevTools disable all but one option.
4. Select the enabled option.
5. Verify the input field's value is the `label` of the selected option.

### CODE REVIEW

File: src/dropdown.ts [hunk 1]
```diff
@@ -318,15 +318,15 @@ export default class Dropdown extends LitElement implements FormControl {
     }
```
Comment (clintcs) [inline src/dropdown.ts:319]: We've now had two bugs where "Select All" found its way into either Dropdown's internal label or its input field. That made me realize there's a structural issue that needed to be addressed. I think I addressed it below by simplifying things quite a bit:

TODO

File: src/dropdown.ts [hunk 2]
```diff
@@ -318,15 +318,15 @@ export default class Dropdown extends LitElement implements FormControl {
...
 
     for (const option of this.#optionElements) {
-      this.#isProgrammaticSelection = true;
+      this.#isSelectionFromValueSetter = true;
```
Comment (clintcs) [inline src/dropdown.ts:321]: This field is only set in one place now. That let me give it a more descriptive name.

File: src/dropdown.ts [hunk 3]
```diff
@@ -1422,6 +1418,7 @@ export default class Dropdown extends LitElement implements FormControl {
       !this.lastSelectedOption.disabled
     ) {
       this.#value = [this.lastSelectedOption.value];
+      this.isShowSingleSelectIcon = Boolean(this.selectedOptions.at(0)?.value);
```
Comment (clintcs) [inline src/dropdown.ts:1421]: If you expand up, you'll see where inside `#onDefaultSlotChange()`.

This was previously set only incidentally via `this.#selectAllElementRef.value.selected = this.isAllSelected`   [above](https://github.com/CrowdStrike/glide-core/pull/903/files#diff-7431ab16692f7d7a7cfecdb04f55d6c62f87417591065650d88dc407a9b62f3bR1409). Setting `selected` on Select All caused `#onOptionsSelectedChange()` to be called, and `#onOptionsSelectedChange()` updated `isShowSingleSelectIcon`.

Now, however, `#onOptionsSelectedChange()` is [only](https://github.com/CrowdStrike/glide-core/pull/903/files#diff-7431ab16692f7d7a7cfecdb04f55d6c62f87417591065650d88dc407a9b62f3bR979) called when `private-selected-change` is dispatched on the default slot and not when it's dispatched by Select All.

That meant that `isShowSingleSelectIcon` was no longer being set both on initial render and when the default slot changed, which caused a test to fail. So I'm `isShowSingleSelectIcon` here, both to fix the failing test and clearer and less error prone to set it directly whenever it needs to be set instead of setting by virtue of something else.

File: src/dropdown.ts [hunk 4]
```diff
@@ -1250,16 +1250,14 @@ export default class Dropdown extends LitElement implements FormControl {
 
   #isOverflowTest = false;
 
-  // Set in the `value` setter and used in `#onOptionsSelectedChange()` to prevent
-  // the latter from adding duplicate values to `value`.
-  //
-  // Also set in `#onDefaultSlotChange()` to prevent `#onOptionsSelectedChange()` from
-  // resetting the user's filter query back to an empty string.
-  #isProgrammaticSelection = false;
```
Comment (clintcs) [inline src/dropdown.ts:1258]: `#isProgrammaticSelection` is now only set in one place. That let me name give it a more descriptive name.

File: src/dropdown.ts [hunk 5]
```diff
@@ -2304,35 +2301,17 @@ export default class Dropdown extends LitElement implements FormControl {
...
-          option !== event.target &&
-          event.target instanceof DropdownOption &&
-          event.target.selected &&
-          option.selected &&
-          event.target !== this.#selectAllElementRef.value
-        ) {
-          option.selected = false;
-        }
-      }
+    if (this.#isSelectionFromSelectAllOrNone) {
```
Comment (clintcs) [inline src/dropdown.ts:2304]: Rather than have guards all over the place below against Select All, we simply return now. `#selectAllOrNone()` now handles setting `#value` for itself.

File: src/dropdown.ts [hunk 6]
```diff
@@ -2304,35 +2301,17 @@ export default class Dropdown extends LitElement implements FormControl {
...
-      event.target !== this.#selectAllElementRef.value &&
-      !this.#isSelectionChangeFromSelectAll &&
-      this.#selectAllElementRef.value
-    ) {
-      // Update Select All to reflect the new selection or deselection.
+    if (this.#selectAllElementRef.value) {
       this.#selectAllElementRef.value.selected = this.isAllSelected;
     }
 
-    this.isShowSingleSelectIcon =
```
Comment (clintcs) [inline src/dropdown.ts:2330]: Moved this inside the conditional below that contains everything else to do with single-select.

File: src/dropdown.ts [hunk 7]
```diff
@@ -2304,35 +2301,17 @@ export default class Dropdown extends LitElement implements FormControl {
   }
 
   #onOptionsSelectedChange(event: Event) {
-    if (!this.multiple) {
-      for (const option of this.#optionElements) {
```
Comment (clintcs) [inline src/dropdown.ts:2308]: Moved this inside the conditional below that contains everything else to do with single-select.

**Review by ynotdraw — APPROVED** [review]
